docs/README.fuse.md

```diff
@@ -0,0 +1,84 @@
+# FUSE Device Plugin
+
+## Introduction
+
+This software is a kubernetes [device plugin](https://kubernetes.io/docs/concepts/cluster-administration/device-plugins/) that exposes /dev/fuse from the system.
+
+## Building
+
+NOTE: This process is not finalized yet.
+
+### Plain binary
+
+```bash
+cd cmd/fuse
+go build .
+```
+
+### Docker
+
+```bash
+docker build -t fuse:latest .
+```
+
+## Running
+
+NOTE: This process is not finalized yet.
+
+### Locally
+```bash
+cd cmd/fuse
+sudo ./fuse -v 3 -logtostderr
+```
+
+### Docker
+```
+docker run -it -v /var/lib/kubelet/device-plugins:/var/lib/kubelet/device-plugins --privileged --cap-add=ALL fuse:latest /bin/bash
+(in docker image) ./fuse -v 3 -logtostderr
+```
+
+## As a DaemonSet
+
+```
+# Deploy the device plugin
+kubectl apply -f manifests/fuse-ds.yml
+```
+
+## API
+
+Node description:
+
+```yaml
+Capacity:
+ ...
+ devices.kubevirt.io/fuse: "3"
+ ...
+```
+
+Pod:
+
+```yaml
+spec:
+  containers:
+  - name: demo
+    ...
+    resources:
+      requests:
+              devices.kubevirt.io/fuse: "1"
+      limits:
+              devices.kubevirt.io/fuse: "1"
+```
+
+The device does not need a priviliged container to use it, but it does require the `mount` system call and CAP_SETUID. The `mount` system call can be made
+available by a seccomp profile.
+
+## Issues
+
+### FUSE Quantity
+
+Currently, Kubernetes device plugins can only expose quantifiable resources.
+FUSE itself is "infinite" resource, so it cannot be properly expressed.
+
+This fact is worked around by the plugin: it adds one more FUSE resource each
```

Comment:
This approach might be raceful. Second VM could request the resource before the number is updated. Consider changing it to hardcoded high-enough amount. There is only limited number of pods you can run on a node anyways.

Reply:
This is equal to the KVM device plugin, after which I modeled this). The number is actually not important really, but it needs to be different for kubernetes to accept it.

Reply:
I can change it to an UUID, but not sure if it brings much

Reply:
I don't believe the KVM DP from this repo is used in KubeVirt.

@stu-gott could you please tell us how does the KVM DP handle its pool of "devices"? Does it use static number of available devices or does it dynamically increase it?